egs/grid/s5/run_dnn.sh

```diff
@@ -0,0 +1,333 @@
+#!/bin/bash
+
+# Copyright 2017  Ruhr-University Bochum (Author: Hendrik Meutzner)
+#
+# Apache 2.0.
+
+# DNN-based audio-visual speech recognition for CHiME-1/2 audio data and GRID video data.
+
+echo ""
+echo "$0"
+date
+
+. ./cmd.sh  # Needed for local or cluster-based processing
+. ./path.sh # Needed for KALDI_ROOT, REC_ROOT and WAV_ROOT
+
+. ./local/check_prerequisites.sh || exit 1
+
+stage=0
+# --stage <stage>       # 0: no skipping (default)
+#                       # 1: skip data preparation
+#                       # 2: skip lang preparation
+#                       # 3: skip feature extraction
+#                       # 4: skip training
+#                       # 5: skip decoding
+#                       # 6: skip DNN training and decoding
+
+# Define number of parallel jobs
+nj=8
+
+eval_list="devel test" # sets used for decoding
+
+# Define feature type
+feat="mfcc"		# MFCC features only
+#feat="fbank"	# Filterbank features only
+#feat="video"	# Video features only
+#feat="av"    # MFCC + video features (using early integration)
+#feat="av2"   # Filterbank + video features (using early integration)
+
+
+# Customize audio data, e.g., setup mixed training.
+# You can provide a list of subfolders for each set (train, devel, test).
+subfolders_train="isolated"
+#subfolders_train="isolated reverberated"
+
+subfolders_test="isolated"
+subfolders_devel="isolated"
+
+# do smbr decoding for DNN experiments (yield somewhat better results but takes pretty long)
+dnn_do_smbr=true
+
+# needed for GMM experiments
+boost_silence=1.0
+
+# ------------------------------------------------------
+# --- script start (no more parameters from here on) ---
+# ------------------------------------------------------
+
+# Parse external parameters into bash variables
+. ./utils/parse_options.sh || exit 1;
+
+# Check for externally provided features
+if [ $# -ge 1 ]; then
+  feat=$1
+fi
+
+# Setup feature file directory
+featdir=$REC_ROOT/feat/$feat
+
+
+# Setup feature directories
+mfcc="$REC_ROOT/feat/mfcc"
+fbank="$REC_ROOT/feat/fbank"
+video="$REC_ROOT/feat/video"
+av="$REC_ROOT/feat/av"
+av2="$REC_ROOT/feat/av2"
+
+# Setup model and decoding directory
+exp="$REC_ROOT/exp"
+mkdir -p $exp
+
+# Setup other relevant directories
+data="$REC_ROOT/data"
+lang="$data/lang"
+dict="$data/local/dict"
+langtmp="$data/local/lang"
+mkdir -p $langtmp
+steps="steps"
+utils="utils"
+
```

Comment:
I don't think the possibility to override the names of steps/ and utils/ directory is really useful, can you please remove it?